### PR TITLE
core: increase precision when summing time deltas in envelope parts

### LINKED ISSUE #7426: core: linear margin: Fail to import train because of discontinuity in the search space
### What happened?

|Train number|Path|Allowance (linear)|RollingStock|
|--------------|--|--------------------|------------|
|9826|LEW MPL|1800s|1TVDUPL|
|5354|LPR NS|1800s|1TVDUPL|
|4758|MSC BX|3600s|22200BM|
|17702| | | |
|4502| | | |
|7834| | | |
|9866| | | |
|9736| | | |
|814322| | | |
|4661| | | |
|4504| | | |
|4306| | | |

### What did you expect to happen?

above trains to be simulated

⚠️ Ok with mareco margins ⚠️ 

### How can we reproduce it (as minimally and precisely as possible)?

use informations in the table to reproduce

### What operating system, browser and environment are you using?

- latest dump from @shenriotpro 

### OSRD version (top right corner `Account` button > `Informations`)

21906b8

### PR DESCRIPTION
**Pull request by eckter**
Fix https://github.com/OpenRailAssociation/osrd/issues/7426

### CODE REVIEW

**Review by multun — APPROVED** [review]

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public final class EnvelopePart implements SearchableEnvelope {
     private double minSpeedCache = Double.NaN;
 
     /** The time from the start of the envelope, in milliseconds. Only read using getTotalTimes. */
-    private long[] cumulativeMSTimesCache = null;
+    private long[] cumulativeMicroSTimesCache = null;
```
Comment (Khoyo) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java]: You missed this comment

**Review by Khoyo — APPROVED** [review]
Otherwise LGTM

**Review by flomonster — APPROVED** [review]

**Comment by multun** [comment]
btw microsecons = us

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ public final class EnvelopePart implements SearchableEnvelope {
     private double minSpeedCache = Double.NaN;
 
     /** The time from the start of the envelope, in milliseconds. Only read using getTotalTimes. */
```
Comment (multun) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java]:
```suggestion
    /** The time from the start of the envelope, in microseconds. Only read using getTotalTimes. */
```

**Pull request merged**